src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBEnumerableValueBinder.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    internal class DocumentDBEnumerableValueBinder<T> : IValueBinder
+    {
+        private DocumentDBContext _context;
+
+        public DocumentDBEnumerableValueBinder(DocumentDBContext context)
+        {
+            _context = context;
+        }
+
+        public Type Type
+        {
+            get
+            {
+                return typeof(IEnumerable<T>);
+            }
+        }
+
+        public object GetValue()
+        {
+            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(_context.ResolvedAttribute.DatabaseName, _context.ResolvedAttribute.CollectionName);
+
+            List<Document> finalResults = new List<Document>();
+
+            string continuation = null;
+
+            do
+            {
+                DocumentQueryResponse response = DocumentDBUtility.RetryAsync(
+                    () => _context.Service.ExecuteNextAsync(collectionUri, _context.ResolvedAttribute.SqlQuery, continuation),
+                    _context.MaxThrottleRetries, codesToIgnore: HttpStatusCode.NotFound).Result;
+
+                if (response == null)
+                {
+                    return null;
+                }
+
+                finalResults.AddRange(response.Results);
+                continuation = response.ResponseContinuation;
+            }
+            while (!string.IsNullOrEmpty(continuation));
```

Comment:
Wondering how Take/Top is handled? E.g. if the user IEnumerable binds to a table with a billion documents, we'll pull all of those into memory? We have some similar issues with our table binding query support (TableAttributeBindingProvider) https://github.com/Azure/azure-webjobs-sdk/issues/903.